tomcat/src/main/java/org/apache/coyote/http11/HttpRequestMessageParser.java

```diff
@@ -0,0 +1,48 @@
+package org.apache.coyote.http11;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class HttpRequestMessageParser {
+
+    private final BufferedReader bufferedReader;
+
+    public HttpRequestMessageParser(final BufferedReader bufferedReader) {
+        this.bufferedReader = bufferedReader;
+    }
+
+    public HttpRequestMessage parseRequestMessage() throws IOException {
+        String startLine = bufferedReader.readLine();
+        if (startLine == null || startLine.isBlank()) {
+            return new HttpRequestMessage(null, null, null);
+        }
+
+        List<String> headers = new ArrayList<>();
+        for (String data = bufferedReader.readLine(); data != null; data = bufferedReader.readLine()) {
+            if (data.isBlank()) {
+                break;
+            }
+            headers.add(data);
+        }
+
+        Map<String, String> headerKeyValue = new HashMap<>();
+        for (int i = 0; i < headers.size(); i++) {
+            final String[] keyValue = headers.get(i).split(": ");
+            headerKeyValue.put(keyValue[0], keyValue[1]);
+        }
+
+        String body = null;
+        if (startLine.startsWith("POST")) {
+            int contentLength = Integer.parseInt(headerKeyValue.get("Content-Length"));
+            char[] buffer = new char[contentLength];
```

Comment:
현재는 모든 데이터가 **텍스트 기반**으로 오겠지만 이전 테스트코드의 코멘트와 같이 **바이너리**로 오게된다면 곤란한 상황이 생길 여지가 있는 것 같아요!

아래쪽 함수들의 파라미터가 전부 `char` 대신 `byte` 를 지원하니 타입을 변환 해보는건 어떨까요?

```suggestion
            byte[] buffer = new byte[contentLength];
```